### PR TITLE
dump: fix the problem that dumpling gets stuck after the database connection is lost

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
fix https://github.com/pingcap/dumpling/issues/243

### What is changed and how it works?
Use `writerCtx` to avoid `taskChan` get blocked after the writer quitted. 

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 
### Release note

- fix the problem that dumpling gets stuck after the database connection is lost

### CODE REVIEW

File: v4/export/sql_type.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ var (
 )
 
 func initColTypeRowReceiverMap() {
+	colTypeRowReceiverMap = map[string]func() RowReceiverStringer{}
```
Comment (overvenus) [inline v4/export/sql_type.go:21]: Dup with line 11?

**Review by overvenus — COMMENTED** [review]
rest lgtm

File: v4/export/sql_type.go (hunk 1 above)
Comment (lichunzhu) [reply v4/export/sql_type.go:21]: I want to clear this map every time we init this.

File: v4/export/dump_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 PingCAP, Inc. Licensed under Apache-2.0.
```
Comment (lance6716) [inline v4/export/dump_test.go:1]:
```suggestion
// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
```

File: v4/export/sql_type.go (hunk 1 above)
Comment (lance6716) [reply v4/export/sql_type.go:21]: currently the only call of `initColTypeRowReceiverMap` is main routine 🤔 Or I guess that you could remove line 11

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: v4/export/sql_type.go (hunk 1 above)
Comment (lichunzhu) [inline v4/export/sql_type.go:21]:
```suggestion
```

**Pull request merged**